packages/fluence-js/src/internal/builtins/SingleModuleSrv.ts

```diff
@@ -0,0 +1,125 @@
+import { v4 as uuidv4 } from 'uuid';
+import { SrvDef } from '../_aqua/single-module-srv';
+import { NodeUtilsDef } from '../_aqua/node-utils';
+import { FluencePeer } from '../FluencePeer';
+import { isNode } from 'browser-or-node';
+import { CallParams } from '../commonTypes';
+import { allowOnlyParticleOriginatedAt, SecurityGuard } from './securityGuard';
+
+export const defaultGuard = (peer: FluencePeer) => {
+    return allowOnlyParticleOriginatedAt<any>(peer.getStatus().peerId!);
+};
+
+export class Srv implements SrvDef {
+    private services: Set<string> = new Set();
+
+    constructor(private peer: FluencePeer) {}
+
+    createSecurityGuard: SecurityGuard<'wasm_b64_content'> = defaultGuard(this.peer);
+
+    async create(wasm_b64_content: string, callParams: CallParams<'wasm_b64_content'>) {
```

Comment:
Hhhm. I've copied the API from rust node. I think we had a reason to do b64s. Anyways, both b64 strings and byte arrays are very inefficient in AVM. So it would better to use IPFS to share big chunks of data.

Reply:
Ok. No problem, thanks. I think it's better for now to have API just like the rust peer